lib/improver/nowcasting/optical_flow.py

```diff
@@ -226,16 +224,25 @@ def point_in_bounds(x, y, nx, ny):
         x_weights = [1. - x_weight_upper, x_weight_upper]
         y_weights = [1. - y_weight_upper, y_weight_upper]
 
+        # Check whether the input data is masked - if so substitute NaNs for
+        # the masked data
+        if isinstance(data, np.ma.MaskedArray):
+            data = np.where(data.mask, np.nan, data.data)
```

Comment:
This includes an implicit type conversion if the input is an integer; np.nan cannot be stored in a integer array, so it becomes a float array. I don't think this is ever going to be an issue, but thought I would mention it for interest.